fhirflat/util.py

```diff
@@ -68,5 +77,132 @@ def get_local_extension_type(t: str):
             raise AttributeError(f"Could not find {t} in fhirflat extensions") from ae
 
 
-def get_local_resource(t: str):
-    return getattr(fhirflat, t)
+def get_local_resource(t: str, case_insensitive: bool = False):
+    if case_insensitive is False:
+        return getattr(fhirflat, t)
+    else:
+        for a in dir(fhirflat):
+            if a.lower() == t.lower():
+                return getattr(fhirflat, a)
+
+
+def find_data_class(
+    data_class: FHIRFlatBase | list[FHIRFlatBase], k: str
+) -> FHIRFlatBase:
+    """
+    Finds the type class for item k within the data class.
+
+    Parameters
+    ----------
+    data_class: list[BaseModel] or BaseModel
+        The data class to search within. If a list, the function will search for the
+        a class with a matching title to k.
+    k: str
+        The property to search for within the data class
+    """
+
+    if isinstance(data_class, list):
+        title_matches = [k.lower() == c.schema()["title"].lower() for c in data_class]
+        result = [x for x, y in zip(data_class, title_matches, strict=True) if y]
+        if len(result) == 1:
+            return get_fhirtype(k)
+        else:
+            raise ValueError(f"Couldn't find a matching class for {k} in {data_class}")
+
+    else:
+        k_schema = data_class.schema()["properties"].get(k)
+
+        base_class = (
+            k_schema.get("items").get("type")
+            if k_schema.get("items") is not None
+            else k_schema.get("type")
+        )
+
+        if base_class is None:
+            assert k_schema.get("type") == "array"
+
+            base_class = [opt.get("type") for opt in k_schema["items"]["anyOf"]]
+        return get_fhirtype(base_class)
+
+
+def code_or_codeable_concept(col_name: str, resource: FHIRFlatBase) -> bool:
+    search_terms = col_name.split(".")
+    fhir_type = find_data_class(resource, search_terms[0])
+
+    if isinstance(fhir_type, list):
+        return code_or_codeable_concept(".".join(search_terms[1:]), fhir_type)
```

Comment:
would this be looping through each fhir_type to see if it is a list? does not match the function def which takes a single resource